var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -605,6 +606,34 @@ def cmake_args(self):
         define = self.define
         define_from_variant = self.define_from_variant
 
+        if self.spec.satisfies("@master:"):
+            ver_in_src = os.path.exists(
+                os.path.join(self.stage.source_path, "packages", "kokkos", "CMakeLists.txt")
+            )
+            if ver_in_src:
+                with open(
+                    os.path.join(self.stage.source_path, "packages", "kokkos", "CMakeLists.txt")
+                ) as f:
+                    all_txt = "".join(f.readlines())
+                    regex = re.compile(r".*set\(Kokkos_VERSION_MAJOR (\d+)")
+                    m_maj = regex.search(all_txt).group(1)
+                    regex = re.compile(r".*set\(Kokkos_VERSION_MINOR (\d+)")
+                    m_min = regex.search(all_txt).group(1)
+                    regex = re.compile(r".*set\(Kokkos_VERSION_PATCH (\d+)")
+                    m_pat = regex.search(all_txt).group(1)
```

Comment:
There's no point in compiling and then searching (just use `re.search(pattern, text)`). I'm also concerned that this is going to be very fragile (what if someone inserts a space, uses capital `set`. And instead of `.*` you should probably have `\s*`; and you could use `Kokkos_VERSION_(MAJOR|MINOR|PATCH)`, etc etc...